test/relayclient/RelayProvider.test.ts

```diff
@@ -0,0 +1,274 @@
+      await stakeManager.stakeForAddress(accounts[1], 1000, {
+        value: ether('1'),
+        from: accounts[2]
+      })
+      await stakeManager.authorizeHub(accounts[1], relayHub.address, { from: accounts[2] })
+      await relayHub.addRelayWorkers([accounts[0]], {
+        from: accounts[1]
+      })
+
+      // create desired transactions
+      misbehavingPaymaster = await TestPaymasterConfigurableMisbehavior.new()
+      await misbehavingPaymaster.setHub(relayHub.address)
+      await misbehavingPaymaster.deposit({ value: web3.utils.toWei('2', 'ether') })
+      const testRecipientForwarderAddress = await testRecipient.getTrustedForwarder()
+      const testRecipientForwarder = await TrustedForwarder.at(testRecipientForwarderAddress)
+      const senderNonce = (await testRecipientForwarder.getNonce(accounts[0])).toString()
+      relayRequest = new RelayRequest({
+        senderAddress: accounts[0],
+        encodedFunction: testRecipient.contract.methods.emitMessage('hello world').encodeABI(),
+        senderNonce,
+        target: testRecipient.address,
+        pctRelayFee: '1',
+        baseRelayFee: '1',
+        gasPrice: '1',
+        gasLimit: '10000',
+        relayWorker: accounts[0],
+        paymaster: misbehavingPaymaster.address
+      })
+      const dataToSign = await getDataToSign({
+        chainId: defaultEnvironment.chainId,
+        verifier: testRecipientForwarderAddress,
+        relayRequest: relayRequest
+      })
+      const signature = await getEip712Signature({
+        web3,
+        dataToSign
+      })
+      await misbehavingPaymaster.setReturnInvalidErrorCode(true)
+      const canRelayFailedReceiptTruffle = await relayHub.relayCall(relayRequest, signature, '0x', {
+        from: accounts[0],
+        gasPrice: '1'
+      })
+      assert.equal(canRelayFailedReceiptTruffle.logs[0].event, 'CanRelayFailed')
+      canRelayFailedTxReceipt = await web3.eth.getTransactionReceipt(canRelayFailedReceiptTruffle.tx)
+
+      await misbehavingPaymaster.setReturnInvalidErrorCode(false)
+      await misbehavingPaymaster.setRevertPreRelayCall(true)
+
+      const innerTxFailedReceiptTruffle = await relayHub.relayCall(relayRequest, signature, '0x', {
+        from: accounts[0],
+        gasPrice: '1'
+      })
+      assert.equal(innerTxFailedReceiptTruffle.logs[0].event, 'TransactionRelayed')
+      assert.equal(innerTxFailedReceiptTruffle.logs[0].args.status.toString(), '2') // PreRelayFailed
+      innerTxFailedReceipt = await web3.eth.getTransactionReceipt(innerTxFailedReceiptTruffle.tx)
+
+      await misbehavingPaymaster.setRevertPreRelayCall(false)
+      const innerTxSuccessReceiptTruffle = await relayHub.relayCall(relayRequest, signature, '0x', {
+        from: accounts[0],
+        gasPrice: '1'
+      })
+      assert.equal(innerTxSuccessReceiptTruffle.logs[0].event, 'SampleRecipientEmitted')
+      assert.equal(innerTxSuccessReceiptTruffle.logs[1].event, 'TransactionRelayed')
```

Comment:
`expectEvent( 'TransacxtionRelayed', { status: '0' } )`
is nicer to read.

Reply:
fixed